lib/pure/coro.nim

```diff
@@ -66,14 +66,18 @@ proc run*() =
   ## Starts main event loop which exits when all coroutines exit. Calling this proc
   ## starts execution of first coroutine.
   var node = coroutines.head
-  var minDelay: float = 0
+  var minDelay: int = 0 # in milliseconds
   var frame: PFrame
   while node != nil:
     var coro = node.value
     current = coro
-    os.sleep(int(minDelay * 1000))
+    os.sleep(minDelay)
 
-    var remaining = coro.sleepTime - (epochTime() - coro.lastRun);
+    # remaining_s - remaining time in seconds
+    let remaining_s = coro.sleepTime - (epochTime() - coro.lastRun)
+    # remaining - in milliseconds
+    # comparing to the int.high is required to avoid integer overflow
```

Comment:
Integer overflow traps in debug builds, you cannot use 'min' to work around this.